### ISSUE TITLE
Explore Proper use of RCs for tC Interop

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/ts-desktop#1228]
Scope out what is required to seamlessly interoperate with tC via the RC spec.

**Comment by neutrinog** [comment]
I agree putting the ts data in .apps an generating a usfm file at the top.

This will require:

1. migrating old ts projects to this new rc structure. We already have a migration workflow. Just need to add another step.
2. Add support for generating the top level usfm file when changes are made to the translation. We already have a regular saving schedule (I think) so we could add a usfm generator to the pipeline.
3. We'll need to update how we read data from the projects. This quickest way would be to move everything "as is" into `.apps/translationStudio/` and update applicable paths in the application. There's a lot we could do to reduce merge conflicts (which has been a big deal for tS) but would probably be better suited under another enhancement.

**Comment by neutrinog** [comment]
The Door43 api is handled by this module https://github.com/unfoldingWord-dev/node-door43-client. We should be able to update this to support the v3 api.

This resource container spec is handled by this module https://github.com/unfoldingWord-dev/node-resource-container. tS is currently using the rc0.1 branch of the code, but there is an rc0.2 branch that may just need a few tweaks in order for us to use it.

**Comment by neutrinog** [comment]
@jag3773 note that in order to provide RC interop with tC we'll need to make tC support RCs. That should only involve updating the manifest so it shouldn't be too invasive. I've added that not the list above.

**Comment by neutrinog** [comment]
- we are not going to update to the v3 api.
- we do not want to update the tc manifest to be rc compliant because we will just have to update to the bible burrito format later this year.
- write up a suggestion on the simplest steps to provide interchangeability tc and ts. This will likely mean we are not going to migrate ts to use the rc spec.

**Comment by neutrinog** [comment]
Here's a simplified list of tasks to accomplish in order to provide tC interop.

1. [ ] **[5]** migrate tS projects to tC the project structure.
1. [ ] **[2]** run project migration during local import.
2. [ ] **[8]** generate USFM file during a save and when exporting/importing.
3. [ ] **[5]** Move all tS project data into `.apps/translationStudio/`
4. [ ] **[5]** Import external edits into the tS project data during an import.

**Comment by neutrinog** [comment]
@jag3773 :point_up:

**Comment by neutrinog** [comment]
Here's a proposal for how to implement the interop https://github.com/unfoldingword-dev/ts-desktop/issues/1243

**Comment by jag3773** [comment]
tS 12.0 has this implement, I'll leave this open until tC implements on it's side.
